testing/functional/tests/cypress/integration/student/02-login-page.spec.ts

```diff
@@ -0,0 +1,220 @@
+import WelcomeObject from "../../page-objects/student-objects/WelcomeObject";
+import LoginObject from "../../page-objects/student-objects/LoginObject";
+
+describe("Login Page", () => {
+  const welcomeObject = new WelcomeObject();
+  const loginObject = new LoginObject();
+
+  const username = Cypress.env("cardSerialNumber");
+  const password = Cypress.env("passcode");
+  const invalidUsername = Cypress.env("invalidCardSerialNumber");
+  const invalidPassword = Cypress.env("invalidPasscode");
+  const caseUsername = Cypress.env("caseSensitivityCardSerialNumber");
+  const fifteenUsername = Cypress.env("fifteenCardSerialNumber");
+
+  beforeEach(() => {
+    cy.visit("/");
+  });
+
+  it("Verify that user able to login with a valid username and valid password.", () => {
+    welcomeObject.virtualTestingButton().should("be.visible").click();
+    welcomeObject.virtualTestingButtonText().should("be.visible");
+    welcomeObject.virtualDeviceId().click({ force: true });
+    cy.wait(2000);
+    loginObject
+      .cardSerialNumberInputText()
+      .type(username)
+      .should("have.value", username);
+    cy.wait(2000);
```

Comment:
I am still missing why we are using the `cy.wait(2000)`. If it is not needed please review all the files and removed them or change them accordingly.

Reply:
### **Cypress offers different ways to the use 'wait'.**

**a) Amount of time to wait in milliseconds.**
    _cy.wait(1000)_

**b) Wait for API response**
_Wait for a specific request to respond, Cypress automatically waits for the network call to complete before proceeding to the next command.
Cypress will wait for all requests to complete within the given requestTimeout and responseTimeout._

**c) Use "defaultCommandTimeout" to change default timeout**
_If 4 seconds are not enough, we can set the time up globally for our project in the cypress.json file to make Cypress wait longer:
{"defaultCommandTimeout": 10000}_

**Using Wait for API response :**
Currently, in our project, I'm using case(b) Wait for API response.
If we are waiting for some resources to be loaded in our website, we can intercept a request and then create an alias for it. That alias will then be used with .wait() command. Test will only continue once that command is finished.
 
<img width="511" alt="image" src="https://user-images.githubusercontent.com/98920571/160001726-3b514f57-e4d0-4fd2-a0a8-c554f968349a.png">

**Using amount of time to wait in milliseconds:**
In this particular case, only I had to use it & removed from other files.

<img width="389" alt="image" src="https://user-images.githubusercontent.com/98920571/160002592-bcbe101b-b768-4a34-b153-92fe283c41b0.png">